test/test_sssd_conf.py

```diff
@@ -0,0 +1,189 @@
+    parser.optionxform = str
+    with config_file.open() as f1:
+        parser.read_file(f1)
+    return parser
+
+
+def parser_object_to_dict(parser_object: configparser.ConfigParser):
+    """
+    Transform content of parser object to dictionary
+
+    :param parser_object: object to be transformed
+    :return: dictionary reflecting content of parser object
+    """
+    content = {section: dict(parser_object.items(section)) for section in parser_object.sections()}
+    return content
+
+
+def compare_conf_files(file1: pathlib.Path, file2: pathlib.Path):
+    """
+    Compare content of two config files ignoring empty lines and reshuffling
+    of content
+
+    :param file1:  first of compared files
+    :param file2:  second of compared files
+    :return: True if content of files matches, False otherwise
+    """
+    dict1 = parser_object_to_dict(load_file_to_parser(file1))
+    dict2 = parser_object_to_dict(load_file_to_parser(file2))
+    assert dict1 == dict2
+
+
+# TESTS
+def test_create_template_exists():
+    """
+    Template file exists
+    """
+    assert SSSDConf._template.is_file()
+
+
+def test_check_backups(tmpdir):
+    """
+    Exception is raised if backup file exists
+    """
+    tmpfile = Path(tmpdir, "testfile")
+    with tmpfile.open('w') as original:
+        original.write('Testing backup original')
+    sssd = SSSDConf()
+    sssd._backup_default = tmpfile
+    sssd._backup_original = tmpfile
+    with pytest.raises(FileExistsError, match=f'{tmpfile} file exists'):
+        sssd.check_backups()
+
+
+def test_create_content_backup_original(sssd_test_prepare):
+    """
+    Backup of original config file is created if the file exists
+    Backup of library default conf_file is created
+    """
+    sssd_test = sssd_test_prepare
+
+    # create conf
+    shutil.copy2(SSSDConf._template, sssd_test._conf_file)
+    sssd_test.create()
+
+    assert sssd_test._backup_original.exists()
+    compare_conf_files(SSSDConf._template, sssd_test._backup_original)
+    assert sssd_test._backup_default.exists()
+    compare_conf_files(SSSDConf._template, sssd_test._backup_default)
+
+
+def test_create_content_backup_default(sssd_test_prepare):
+    """
+    Backup of library default config file is always created
+    regardless of previous existence of config file
+    """
+    sssd_test = sssd_test_prepare
+
+    sssd_test.create()
+    assert sssd_test._backup_default.exists()
+    compare_conf_files(SSSDConf._template, sssd_test._backup_default)
+
+
+def test_create_content(sssd_test_prepare):
+    """
+    Content of config file is created based on template if the file is not
+    present in the system. Content of created file corresponds to template values.
+    """
+    sssd_test = sssd_test_prepare
+
+    sssd_test.create()
+
+    # content of created config is saved to backup_default and therefore
+    # it can be verified as follows
+    compare_conf_files(SSSDConf._template, sssd_test._backup_default)
```

Comment:
Not sure we need this test when we have previous test that is shaking the same thing

Reply:
Agree, removing this one.